### ISSUE TITLE
Segmentation fault when using onHandShake with 1.8.8

### ISSUE THREAD

**Issue by garveen** [swoole/swoole-src#808]
env: alpine 3.4 on docker
php: 7.0.8

swoole-1.8.7:
```
lstat("/DATA/vendor/composer/../acabin/laravoole", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/DATA/vendor/composer/../acabin", {st_mode=S_IFDIR|0755, st_size=0, ...}) = 0
lstat("/DATA/vendor/composer", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/DATA/vendor", {st_mode=S_IFDIR|0755, st_size=8192, ...}) = 0
lstat("/DATA", {st_mode=S_IFDIR|0755, st_size=8192, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Wrapper/SwooleWebSocketWrapper.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=5006, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=5006, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=5006, ...}) = 0
mmap(NULL, 5006, PROT_READ, MAP_SHARED, 3, 0) = 0x7f75f8405000
munmap(0x7f75f8405000, 5006)            = 0
close(3)                                = 0
access("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/SwooleHttpWrapper.php", F_OK) = 0
clock_gettime(CLOCK_REALTIME, {1470910983, 86962900}) = 0
lstat("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/SwooleHttpWrapper.php", {st_mode=S_IFREG|0755, st_size=1589, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Wrapper/SwooleHttpWrapper.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=1589, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=1589, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=1589, ...}) = 0
mmap(NULL, 1589, PROT_READ, MAP_SHARED, 3, 0) = 0x7f75f8406000
munmap(0x7f75f8406000, 1589)            = 0
close(3)                                = 0
access("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/Swoole.php", F_OK) = 0
clock_gettime(CLOCK_REALTIME, {1470910983, 97827400}) = 0
lstat("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/Swoole.php", {st_mode=S_IFREG|0755, st_size=2739, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Wrapper/Swoole.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=2739, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=2739, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=2739, ...}) = 0
mmap(NULL, 2739, PROT_READ, MAP_SHARED, 3, 0) = 0x7f75f8406000
munmap(0x7f75f8406000, 2739)            = 0
close(3)                                = 0
access("/DATA/vendor/composer/../acabin/laravoole/src/Base.php", F_OK) = 0
clock_gettime(CLOCK_REALTIME, {1470910983, 108411600}) = 0
lstat("/DATA/vendor/composer/../acabin/laravoole/src/Base.php", {st_mode=S_IFREG|0755, st_size=5912, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Base.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=5912, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=5912, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=5912, ...}) = 0
mmap(NULL, 5912, PROT_READ, MAP_SHARED, 3, 0) = 0x7f75f8405000
munmap(0x7f75f8405000, 5912)            = 0
close(3)                                = 0
access("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/ServerInterface.php", F_OK) = 0
clock_gettime(CLOCK_REALTIME, {1470910983, 113622000}) = 0
lstat("/DATA/vendor/composer/../acabin/laravoole/src/Wrapper/ServerInterface.php", {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Wrapper/ServerInterface.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
mmap(NULL, 547, PROT_READ, MAP_SHARED, 3, 0) = 0x7f75f8406000
munmap(0x7f75f8406000, 547)             = 0
close(3)                                = 0
pipe2([3, 4], O_CLOEXEC)                = 0
ioctl(4, TIOCGWINSZ, 0x7ffd8133bbd8)    = -1 ENOTTY (Not a tty)
pipe2([5, 6], O_CLOEXEC)                = 0
rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
clone(child_stack=0x7ffd8133bb98, flags=CLONE_VM|CLONE_VFORK|SIGCHLD) = 22842
close(6)                                = 0
read(5, "", 4)                          = 0
close(5)                                = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
fcntl(4, F_SETFD, 0)                    = 0
close(3)                                = 0
write(4, "a:3:{s:6:\"config\";a:9:{s:4:\"host"..., 551) = 551
close(4)                                = 0
wait4(22842,
```

swoole-1.8.8:

```
lstat("/DATA/vendor/composer/../acabin/laravoole/src", {st_mode=S_IFDIR|0755, st_size=8192, ...}) = 0
lstat("/DATA/vendor/composer/../acabin/laravoole", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/DATA/vendor/composer/../acabin", {st_mode=S_IFDIR|0755, st_size=0, ...}) = 0
open("/DATA/vendor/acabin/laravoole/src/Wrapper/ServerInterface.php", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
fstat(3, {st_mode=S_IFREG|0755, st_size=547, ...}) = 0
mmap(NULL, 547, PROT_READ, MAP_SHARED, 3, 0) = 0x7f2906a15000
munmap(0x7f2906a15000, 547)             = 0
close(3)                                = 0
pipe2([3, 4], O_CLOEXEC)                = 0
ioctl(4, TIOCGWINSZ, 0x7ffcca957f08)    = -1 ENOTTY (Not a tty)
pipe2([5, 6], O_CLOEXEC)                = 0
rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
clone(child_stack=0x7ffcca957ec8, flags=CLONE_VM|CLONE_VFORK|SIGCHLD) = 22
close(6)                                = 0
read(5, "", 4)                          = 0
close(5)                                = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
fcntl(4, F_SETFD, 0)                    = 0
close(3)                                = 0
write(4, "a:3:{s:6:\"config\";a:9:{s:4:\"host"..., 551) = 551
close(4)                                = 0
wait4(22, Segmentation fault
[{WIFEXITED(s) && WEXITSTATUS(s) == 139}], 0, NULL) = 22
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=22, si_uid=0, si_status=139, si_utime=0, si_stime=0} ---
close(2)                                = 0
close(1)                                = 0
close(0)                                = 0
munmap(0x7f2903400000, 2097152)         = 0
munmap(0x7f2906a16000, 65536)           = 0
munmap(0x7f290691b000, 593920)          = 0
madvise(0x5556403e1000, 389120, MADV_DONTNEED) = 0
munmap(0x7f2903600000, 2097152)         = 0
munmap(0x7f2905000000, 2097152)         = 0
munmap(0x7f2906aa7000, 323584)          = 0
exit_group(0)                           = ?
+++ exited with 0 +++






bash-4.3# php --ri swoole

swoole

swoole support => enabled
Version => 1.8.8
Author => tianfeng.han[email: mikan.tenny@gmail.com]
epoll => enabled
eventfd => enabled
timerfd => enabled
signalfd => enabled
spinlock => enabled
rwlock => enabled
async http/websocket client => enabled
Linux Native AIO => enabled
zlib => enabled
mutex_timedlock => enabled
pthread_barrier => enabled

Directive => Local Value => Master Value
swoole.aio_thread_num => 2 => 2
swoole.display_errors => On => On
swoole.use_namespace => Off => Off
swoole.message_queue_key => 0 => 0
swoole.unixsock_buffer_size => 8388608 => 8388608
```

**Comment by garveen** [comment]
PS: Only seg fault with onHandShake

**Comment by matyhtf** [comment]
This bug has been fixed in current branch.

**Comment by garveen** [comment]
Because this is a critical bug, should there be a hotfix version like 1.8.9 be released? 1.8.8 was a 15-days-old version.
